java/team25core/VuMarkIdentificationTask.java

```diff
@@ -0,0 +1,126 @@
+        CENTER,
+        UNKNOWN,
+    }
+
+    public enum PollingMode {
+        ON,
+        OFF,
+    }
+
+    public class VuMarkIdentificationEvent extends RobotEvent
+    {
+        public EventKind kind;
+
+        public VuMarkIdentificationEvent(RobotTask task, EventKind kind)
+        {
+            super(task);
+            this.kind = kind;
+        }
+        public String toString()
+        {
+            return kind.toString();
+        }
+    }
+
+    protected VuforiaLocalizerCustom vuforia;
+    protected VuforiaTrackable relicTemplate;
+    protected Telemetry.Item vuMarkTelemetry;
+    protected ElapsedTime pollTimer;
+    protected PollingMode pollingMode;
+    protected RelicRecoveryVuMark glyphPosition;
+
+    protected final static int POLL_RATE = 1;
+
+    public VuMarkIdentificationTask(Robot robot)
+    {
+        super(robot);
+        this.vuMarkTelemetry = robot.telemetry.addData("Vumark: ", "Not Visible");
+        this.pollingMode = PollingMode.OFF;
+    }
+
+    public VuMarkIdentificationTask(Robot robot, VuforiaLocalizerCustom vuforia)
+    {
+        super(robot);
+        this.vuforia = vuforia;
+        this.vuMarkTelemetry = robot.telemetry.addData("Vumark: ", "Not Visible");
+        this.pollingMode = PollingMode.OFF;
+    }
+
+    @Override
+    public void start()
+    {
+        VuforiaTrackables relicTrackables = this.vuforia.loadTrackablesFromAsset("RelicVuMark");
+        relicTemplate = relicTrackables.get(0);
+
+        relicTemplate.setName("relicVuMarkTemplate"); // can help in debugging; otherwise not necessary
+        relicTrackables.activate();
+    }
+
+    @Override
+    public void stop()
+    {
+
+    }
+
+    public void setPollingMode(PollingMode pollingMode)
+    {
+        this.pollingMode = pollingMode;
+
+        if (pollingMode == PollingMode.ON) {
+            pollTimer = new ElapsedTime(ElapsedTime.Resolution.SECONDS);
+        }
+    }
+
+    public RelicRecoveryVuMark getGlyphPosition() {
+
+        return this.glyphPosition;
+    }
+
+    @Override
+    public boolean timeslice()
+    {
+        if ((pollingMode == PollingMode.ON) && (pollTimer.time() > POLL_RATE)) {
+            vuforia.forceRefreshBitmap();
+            pollTimer.reset();
+            return false;
+        }
+        /**
+         * See if any of the instances of {@link relicTemplate} are currently visible.
+         * {@link RelicRecoveryVuMark} is an enum which can have the following values:
+         * UNKNOWN, LEFT, CENTER, and RIGHT. When a VuMark is visible, something other than
+         * UNKNOWN will be returned by {@link RelicRecoveryVuMark#from(VuforiaTrackable)}.
+         */
+        RelicRecoveryVuMark vuMark = RelicRecoveryVuMark.from(relicTemplate);
+        if (vuMark != RelicRecoveryVuMark.UNKNOWN) {
+           // vuMarkTelemetry.addData("VuMark", "%s visible", vuMark);
+            vuMarkTelemetry.setValue("VuMark: %s visible", vuMark.toString());
+            glyphPosition = vuMark;
```

Comment:
Hmm, well this may work, but doesn't exactly fit within the task/event framework architecture.

You should be queuing a VuMarkIdentificationEvent here.  And getGlyphPosition() goes away entirely.  And in your autonomous, you have an event handler that deals with the VuMarkIdentificationEvent events.  For an example of what I'm talking about take a very close look at the event creation in ColorThiefTask's timeslice.

Can you please make these changes and re-push?